mlfinlab/tests/test_sb_bagging.py

```diff
@@ -0,0 +1,310 @@
+                                       class_weight='balanced_subsample', max_depth=12)
+        clf_3 = KNeighborsClassifier()
+        clf_4 = LinearSVC()
+
+        sb_clf_1 = SequentiallyBootstrappedBaggingClassifier(base_estimator=clf_1, max_features=0.2,
+                                                             n_estimators=100,
+                                                             triple_barrier_events=self.meta_labeled_events,
+                                                             price_bars=self.data, oob_score=True,
+                                                             random_state=1, bootstrap_features=True,
+                                                             max_samples=30, verbose=2)
+
+        sb_clf_2 = SequentiallyBootstrappedBaggingClassifier(base_estimator=clf_2, max_features=7,
+                                                             n_estimators=100,
+                                                             triple_barrier_events=self.meta_labeled_events,
+                                                             price_bars=self.data, oob_score=False,
+                                                             random_state=1, bootstrap_features=True,
+                                                             max_samples=0.3, warm_start=True)
+
+        sb_clf_3 = SequentiallyBootstrappedBaggingClassifier(base_estimator=clf_3,
+                                                             triple_barrier_events=self.meta_labeled_events,
+                                                             price_bars=self.data)
+
+        sb_clf_4 = SequentiallyBootstrappedBaggingClassifier(base_estimator=clf_4, max_features=0.2,
+                                                             n_estimators=100,
+                                                             triple_barrier_events=self.meta_labeled_events,
+                                                             price_bars=self.data, oob_score=True,
+                                                             random_state=1, bootstrap_features=True,
+                                                             max_samples=30, verbose=2)
+
+        sb_clf_1.fit(self.X_train, self.y_train_clf, sample_weight=np.ones((self.X_train.shape[0],)), )
+
+        sb_clf_2.fit(self.X_train, self.y_train_clf, sample_weight=np.ones((self.X_train.shape[0],)), )
+        sb_clf_2.n_estimators += 0
+        sb_clf_2.fit(self.X_train, self.y_train_clf, sample_weight=np.ones((self.X_train.shape[0],)), )
+        sb_clf_2.n_estimators += 2
+        sb_clf_2.fit(self.X_train, self.y_train_clf, sample_weight=np.ones((self.X_train.shape[0],)), )
+
+        sb_clf_3.fit(self.X_train, self.y_train_clf)
+        sb_clf_4.fit(self.X_train, self.y_train_clf)
+
+    def test_value_error_raise(self):
+        """
+        Test various values error raise
+        """
+        clf = KNeighborsClassifier()
```

Comment:
Can we add inline comments on what the different value errors are which the test is attempting?

Reply:
Sure